### DISCUSSION TITLE
Issue viewing Postgres data in production

### DISCUSSION THREAD

**Discussion by domwakeling** [vercel/community#3590] [Help] [answered]
### Summary

I have a new Nextjs project that is using Vercel Postgres (currently two tables and circa 30 rows). It is working fine locally; however it appears that whenever I push a change to production all existing rows are dropped and I am left with two empty tables. This clearly can't be the intended effect!

### Steps to Reproduce

Data populated; GET call to API locally ... 
![Screenshot 2023-08-13 at 10 38 38](https://github.com/vercel/community/assets/11430006/6774144d-9217-43f3-afe9-f3bf83f6a373)

... and same route on the Vercel version ...
![Screenshot 2023-08-13 at 10 40 42](https://github.com/vercel/community/assets/11430006/dac4b300-563b-44f6-801c-cb5bcbcc132b)

Then made a change to the README, commited and pushed. Once deployed to Vercel I get:
![Screenshot 2023-08-13 at 10 42 20](https://github.com/vercel/community/assets/11430006/8d611a89-5b6f-4920-9647-956a5cbd03cb)
(checking in the project dashboard, the tables still exist but there are no rows)

Followed the [setup instructions](https://vercel.com/docs/storage/vercel-postgres/quickstart) (other than not using the Vercel CLI - appears to 'only' be adding environment variables to a local `.env` which I did myself) and at a loss as to what's wrong.

**Comment by leerob** [comment, marked as answer]
Vercel Postgres is just open-source Postgres, so one way to verify whether this is application code or something with your configuration on Vercel would be to use a local Docker instance of Postgres. It's hard to tell what is happening with more code here. Especially since you said:

> Checking in the project dashboard, the tables still exist but there are no rows

That sounds like an application error in your API.

- Have you looked at the [logs in your application](https://vercel.com/docs/concepts/observability/runtime-logs)?
- Are you sure the [environment variables](https://vercel.com/docs/concepts/projects/environment-variables) are added to the correct project?

**Reply by domwakeling** [reply]
I have the environment variables set (I have a feeling that they were automatically added to the Vercel project without my intervention, but they're definitely there) - the "production" version on Vercel can read/write to it just fine.

Checking deploy logs (yes, should have done that sooner) - there is a server component that puts in two calls; the first is to a "manufacturers" table and the second is a join on "ashps" and "manufacturers". From the logs it looks as though the initial call to mannufacturers is successful (can't tell whether it gets any data, ie if the rows are still there) and the second (joined) call is where things are going wrong.

The calls don't cause a problem in local development; and if I re-populate the data they don't cause a problem on subsequent server calls/revalidations etc.

Is there anything special happening on deployment after a GitHub change that doesn't happen any other time?!

![Screenshot 2023-08-13 at 23 57 27](https://github.com/vercel/community/assets/11430006/7237d9b2-8a2c-4f59-8af9-4a65ebb7236a)

**Reply by domwakeling** [reply]
Follow-up: since the only time this seems to happen is when I push a change to GitHub, not sure how a local Docker instance of Postgres would help?

**Reply by domwakeling** [reply]
Identified issue running `next build` locally - thanks
